packages/website/docs/sources.md

```diff
@@ -236,25 +260,158 @@ const autocomplete = createAutocomplete({
+
+:::
+
+### Customizing items with templates
+
+In addition to defining data sources for items, a source also lets you customize how to display items using [`templates`](#templates-specific-to-algoliaautocomplete-js).
+
+:::info
+
+Templates are only supported in [autocomplete-js](autocomplete-js).
+
+:::
+
+Templates can return a string:
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return `Result: ${item.name}`;
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+Or a [Preact](https://preactjs.com/) component:
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+import { h } from 'preact';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return h('div', null, item.name);
+          },
+        },
+      },
+    ];
+  },
+});
+
+```
+
+Or HTML/JSX-like syntax (here using [htm](https://github.com/developit/htm)):
+
+```js
+import { autocomplete, highlightHit } from '@algolia/autocomplete-js';
+import { html } from 'htm/preact';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return html`<div>
+              ${html([highlightHit({ hit: item, attribute: 'name' })])}
+            </div>`
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+Or a custom component using `createElement` and `Fragment`:
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item, createElement, Fragment }) {
+            return createElement(Fragment, {}, item.name);
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+You can also define `header` and `footer` templates to display before and after the items list.
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          header() {
+            return 'Suggestions';
+          },
+          // ...
+          footer() {
+            return 'Footer';
+          },
+        },
+      },
+    ];
+  },
+});
+```
 
 ## Reference
 
 ### `getSources`
 
 > `(params: { query: string, state: AutocompleteState, ...setters: Autocomplete Setters }) => Array<AutocompleteSource> | Promise<Array<AutocompleteSource>>`
 
-The function to fetch the sources and their behaviors.
+The function to fetch sources and their behaviors.
 
-A source is described by the following properties:
+A source implements the following interface:
 
 ### `getItems`
```

Comment:
Theses should be level 4 (`####`) in terms of hierarchy, but they would then be hidden  by in the TOC by Docusaurus. Any recommendations?

Reply:
I would document the two sets separately [like here](https://deploy-preview-396--algolia-autocomplete.netlify.app/docs/state), one being "Sources" and the other "Source". This way we can clarify they aren't at the same level, yet keep them as level-3 headings.

Another way is to document them in the form `getSources[].getItems` or `getSources -> getItems` but I personally find it unsightly, idiosyncratic, and confusing.